controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -332,38 +331,41 @@ void master_thread::handle_cmdu_1905_autoconfiguration_search(Socket *sd,
     }
     default: {
         LOG(ERROR) << "unknown autoconfig freq band, value=" << int(auto_config_freq_band);
-        return;
+        return false;
     }
     }
 
     auto tlvSupportedServiceOut = cmdu_tx.addClass<wfa_map::tlvSupportedService>();
     if (!tlvSupportedServiceOut) {
         LOG(ERROR) << "addClass wfa_map::tlvSupportedService failed";
-        return;
+        return false;
     }
     if (!tlvSupportedServiceOut->alloc_supported_service_list()) {
         LOG(ERROR) << "alloc_supported_service_list failed";
-        return;
+        return false;
     }
     auto supportedServiceTuple = tlvSupportedServiceOut->supported_service_list(0);
     if (!std::get<0>(supportedServiceTuple)) {
         LOG(ERROR) << "Failed accessing supported_service_list";
-        return;
+        return false;
     }
     std::get<1>(supportedServiceTuple) =
         wfa_map::tlvSupportedService::eSupportedService::MULTI_AP_CONTROLLER;
 
     LOG(DEBUG) << "sending autoconfig response message";
-    son_actions::send_cmdu_to_agent(sd, cmdu_tx);
-    LOG(DEBUG) << "sent";
+    return son_actions::send_cmdu_to_agent(sd, cmdu_tx);
 }
 
 bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(Socket *sd,
                                                               ieee1905_1::CmduMessageRx &cmdu_rx)
 {
     auto tlvAp = cmdu_rx.addClass<wfa_map::tlvApRadioBasicCapabilities>();
     if (tlvAp == nullptr) {
-        LOG(ERROR) << "Failed to get APRadioBasicCapabilities";
+        // Check if this a message we sent to the agent which was just looped back
```

Comment:
This needs a better comment... You need to explain why it would be looped back in the first place, and why you can use the radio identifier to detect loopback (which you can't, actually, see below). Something like:

```
// Check if this is a M2 message that we sent to the agent, which was just looped back.
// The M1 and M2 messages are both of CMDU type AP_Autoconfiguration_WSC. Thus,
// when we send the M2 to the local agent, it will be published back on the local bus because
// the destination is our AL-MAC, and the controller does listen to this CMDU.
// Ideally, we should use the message type attribute from the WSC payload to distinguish.
// Unfortunately, that is a bit complicated with the current tlv parser. However, there is another
// way to distinguish them: the M1 message has the AP_Radio_Basic_Capabilities TLV,
// while the M2 has the AP_Radio_Identifier TLV.
// If this is a looped back M2 CMDU, we can treat is as handled successfully.
```

Unfortunately this distinction is not entirely correct, because the M1 may come from a non-MultiAP 1905.1 device. In that case, there is no AP Radio Basic Capabilities in M1 and no AP Radio Identifier in M2. But that is not something we have to handle now. :ok: